codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -833,34 +849,56 @@ impl Event {
     }
 
     pub fn from_evm_event_config(
-        human_cfg_event: EventConfig,
+        event_config: EvmEventConfig,
         opt_abi: &Option<EvmAbi>,
     ) -> Result<Self> {
         let event: NormalizedEthAbiEvent =
-            Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
+            Event::get_abi_event(&event_config.event, opt_abi)?.into();
+        let topic0 = ethers::core::utils::hex::encode_prefixed(ethers::utils::keccak256(
+            event.0.abi_signature().as_bytes(),
+        ));
 
         Ok(Event {
-            name: human_cfg_event.name.unwrap_or(event.0.name.to_owned()),
+            name: event_config.name.unwrap_or(event.0.name.to_owned()),
+            payload: EventPayload::Params(event.0.inputs.clone()),
             event,
+            topic0,
         })
     }
 
-    fn get_event(&self) -> &EthAbiEvent {
-        &self.event.0
-    }
+    // TODO: Validatate that all event names are unique
+    // TODO: Validatate that event names are not reserved
+    pub fn from_fuel_event_config(
+        event_config: FuelEventConfig,
+        fuel_abi: &FuelAbi,
+    ) -> Result<Self> {
+        let event: NormalizedEthAbiEvent =
+            Event::get_abi_event(&format!("{}()", event_config.name), &None)?.into();
```

Comment:
Cool so this is just a stub with an empty event for now? Need to relook at the data structure.

Reply:
Unfortunately it's still used in some places, so I couldn't get rid of it yet.